src/Data/List/Lazy.purs

```diff
@@ -295,6 +320,29 @@ index xs = go (step xs)
 -- | An infix synonym for `index`.
 infixl 8 index as !!
 
+-- | Find the index of the first element equal to the specified element.
+elemIndex :: forall a. Eq a => a -> List a -> Maybe Int
+elemIndex x = findIndex (_ == x)
+
+-- | Find the index of the last element equal to the specified element.
+elemLastIndex :: forall a. Eq a => a -> List a -> Maybe Int
+elemLastIndex x = findLastIndex (_ == x)
+
+-- | Find the first index for which a predicate holds.
+findIndex :: forall a. (a -> Boolean) -> List a -> Maybe Int
+findIndex fn = go 0
+  where
+  go :: Int -> List a -> Maybe Int
+  go n list = do
+      o <- uncons list 
+      if fn o.head 
+         then pure n 
+         else go (n + 1) o.tail
+
+-- | Find the last index for which a predicate holds.
+findLastIndex :: forall a. (a -> Boolean) -> List a -> Maybe Int
+findLastIndex fn xs = ((length xs - 1) -) <$> findIndex fn (reverse xs)
```

Comment:
This doesn't parse with the current 0.9 branch due to the operator section, it needs to be `((length xs - 1) - _)` now.